supervisor/options.py

```diff
@@ -1395,7 +1392,11 @@ def drop_privileges(self, user):
             os.setgid(gid)
         except OSError:
             return 'Could not set group id of effective user'
-        os.setuid(uid)
+
+        try:
+            os.setuid(uid)
+        except:
```

Comment:
This should be `except OSError` like the others.  

You added this `try`/`except` around `os.setuid()`.  Previously, there was none and this function would throw an exception from `os.setuid()`.  After this change, it will no longer do this.  Have you thoroughly checked that every possible code path that calls this function (`drop_privileges`) will not change behavior as a result of this change?

This change seems more consistent with the rest of the function, but it can't be made without carefully considering its impact on the rest of the program.